data-service/src/test/java/com/amazonaws/blox/dataservice/api/DescribeEnvironmentApiTest.java

```diff
@@ -0,0 +1,144 @@
+            .deploymentConfiguration(new DeploymentConfiguration())
+            .deploymentMethod(DEPLOYMENT_METHOD)
+            .validEnvironment(true)
+            .build();
+    describeEnvironmentRequest =
+        DescribeEnvironmentRequest.builder().environmentId(environmentIdWrapper).build();
+  }
+
+  @Test
+  public void describeEnvironmentSuccess() throws Exception {
+    when(environmentRepository.getEnvironment(environmentId)).thenReturn(environment);
+    final DescribeEnvironmentResponse describeEnvironmentResponse =
+        describeEnvironmentApi.describeEnvironment(describeEnvironmentRequest);
+
+    verify(environmentRepository).getEnvironment(environmentId);
+
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getEnvironmentId(), environmentIdWrapper);
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getEnvironmentHealth(),
+        environment.getEnvironmentHealth().name());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getActiveEnvironmentRevisionId(),
+        environment.getActiveEnvironmentRevisionId());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getCreatedTime(),
+        environment.getCreatedTime());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getEnvironmentStatus(),
+        environment.getEnvironmentStatus().name());
+    assertEquals(describeEnvironmentResponse.getEnvironment().getRole(), environment.getRole());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getDeploymentConfiguration(),
+        apiModelMapper.toWrapperDeploymentConfiguration(environment.getDeploymentConfiguration()));
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getEnvironmentType().name(),
+        environment.getEnvironmentType().name());
+    assertEquals(
+        describeEnvironmentResponse.getEnvironment().getLastUpdatedTime(),
+        environment.getLastUpdatedTime());
+  }
+
+  @Test(expected = ResourceNotFoundException.class)
+  public void describeEnvironmentResourceNotFoundException() throws Exception {
+    when(environmentRepository.getEnvironment(isA(EnvironmentId.class)))
+        .thenThrow(new ResourceNotFoundException(ResourceType.ENVIRONMENT, ENVIRONMENT_NAME));
+    describeEnvironmentApi.describeEnvironment(describeEnvironmentRequest);
+  }
+
+  @Test(expected = InternalServiceException.class)
+  public void describeEnvironmentInternalServiceException() throws Exception {
+    when(environmentRepository.getEnvironment(isA(EnvironmentId.class)))
+        .thenThrow(new InternalServiceException(""));
+    describeEnvironmentApi.describeEnvironment(describeEnvironmentRequest);
+  }
+
+  @Test(expected = InternalServiceException.class)
+  public void describeEnvironmentInternalServiceExceptionWithUnknownException() throws Exception {
+    when(environmentRepository.getEnvironment(isA(EnvironmentId.class))).thenReturn(null);
```

Comment:
`Return(null)` looks not intuitive to understand what this test about, kind of hack as following code will cause NPE.   Would be good to use `when(environmentRepository.getEnvironment(isA(EnvironmentId.class))).thenThrow(new IllegalStateException(""))`

Reply:
I was thinking of letting environmentRepository.getEnvironment() throw some other exceptions. Just not sure if we should also test the exceptions that are not from environmentRepository.getEnvironment(). 
I can keep both unit test cases and rename them into something more intuitive like

describeEnvironmentUnknownException()

and 

describeEnvironmentOtherExceptions() ?